service_provider.go

```diff
@@ -933,4 +933,107 @@ func (sp *ServiceProvider) validateLogoutResponse(resp *LogoutResponse) error {
 	}
 	
 	return nil
-}
\ No newline at end of file
+}
+
+// ValidateLogoutRequestRequest validates the LogoutRequest content from the request
+func (sp *ServiceProvider) ValidateLogoutRequestRequest(req *http.Request) error {
+	if data := req.URL.Query().Get("SAMLRequest"); data != "" {
+		return sp.ValidateLogoutRequestRedirect(data)
+	}
+
+	err := req.ParseForm()
+	if err != nil {
+		return fmt.Errorf("unable to parse form: %v", err)
+	}
+
+	return sp.ValidateLogoutRequestForm(req.PostForm.Get("SAMLRequest"))
+}
+
+// ValidatePostLogoutRequest returns a nil error if the logout response is valid.
+func (sp *ServiceProvider) ValidateLogoutRequestForm(postFormData string) error {
+	rawRequestBuf, err := base64.StdEncoding.DecodeString(postFormData)
+	if err != nil {
+		return fmt.Errorf("unable to parse base64: %s", err)
+	}
+
+	var resp LogoutRequest
+
+	if err := xml.Unmarshal(rawRequestBuf, &resp); err != nil {
+		return fmt.Errorf("cannot unmarshal response: %s", err)
+	}
+
+	if err := sp.validateLogoutRequest(&resp); err != nil {
+		return err
+	}
+
+	doc := etree.NewDocument()
+	if err := doc.ReadFromBytes(rawRequestBuf); err != nil {
+		return err
+	}
+
+	requestEl := doc.Root()
+	if err = sp.validateSigned(requestEl); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ValidateRedirectLogoutRequest returns a nil error if the logout response is valid.
+// URL Binding appears to be gzip / flate encoded
+// See https://www.oasis-open.org/committees/download.php/20645/sstc-saml-tech-overview-2%200-draft-10.pdf  6.6
+func (sp *ServiceProvider) ValidateLogoutRequestRedirect(queryParameterData string) error {
+	rawRequestBuf, err := base64.StdEncoding.DecodeString(queryParameterData)
+	if err != nil {
+		return fmt.Errorf("unable to parse base64: %s", err)
+	}
+
+	gr := flate.NewReader(bytes.NewBuffer(rawRequestBuf))
+
+	decoder := xml.NewDecoder(gr)
+
+	var resp LogoutRequest
+
+	err = decoder.Decode(&resp)
+	if err != nil {
+		return fmt.Errorf("unable to flate decode: %s", err)
+	}
+
+	if err := sp.validateLogoutRequest(&resp); err != nil {
+		return err
+	}
+
+	doc := etree.NewDocument()
+	if _, err := doc.ReadFrom(gr); err != nil {
+		return err
+	}
+
+	responseEl := doc.Root()
+	if responseEl == nil {
```

Comment:
I'm not sure I understand your TODO here and why responseEl might be nil. 

My main concern is that there is a risk we are skipping validation and returning a nil error in a case where the caller might think the signature was actually validated?

If this is all fine, could you please add/improve the comment so it's more clear why this code path is ok?

Reply:
Okay, so now I see that this is addressed by #253 / #254. 

I think that we should resolve this issue before this PR lands, because I don't want to skip signature validation in a way that the caller can't easily recognize.

If there is some reason to merge this without fixing, then perhaps we can return a structured error that the caller can detect. Something like `return ErrDetachedSignatureNotYetImplemented` or what have you.

Reply:
Yes that makes sense, thank you I'll look how to verify the signature!